packages/tooling/fast-tooling-react/src/form/custom-controls/control.css.md

```diff
@@ -0,0 +1,122 @@
+# Abstract
+
+The CSS controls consist of an abstract control and two extended controls, inline CSS and internal CSS. These controls will provide a way to interpret CSS strings for inline and internal CSS into form UI and back to strings.
+
+## Build script
+
+The package should have a `devDependency` for [MDN data](https://www.npmjs.com/package/mdn-data) and re-interpret the properties based on this. To only require what is needed there should be a build script which takes the MDN data and converts it to a `.ts` file with exports conforming to an interface which defines the example provided in [Variables](#variables).
+
+## Variables
+
+There should be a list of internal variables that are assigned to each CSS property. The structure should look like this:
+
+```typescript
+[
+    {
+        "border": { // The property name should be pascalCase
+            name: "border", // The name should be spinal-case
+            default: "none",
+            value: [
+                {
+                    type: "width",
+                    default: "1px"
+                },
+                {
+                    type: "border-style",
+                    default: "solid"
```

Comment:
Did you consider mapping more closely to the CSS properties themselves rather than the shorthand? "border" vs "border-style", "border-color", "border-width" or even "border-left-color", "border-left-style", "border-left-width".

Thinking out loud here - mostly just asking if this is going to provide all the flexibility needed because CSS shorthand gets a bit complicated, with argument count and argument order mattering.

Reply:
The generated UI should reflect the shorthand, but we could do it that way where we just have "border-width", "border-style", "border-color" in a sort of logical "grouping". I considered allowing everything and then having some method of configuration so that "border-left-style" would be omitted for the sake of brevity.

Reply:
I've updated to make `shorthand` CSS properties the default, if a user wants to set it so that they get all CSS properties they can set `shorthandOnly` to `false`. What we're trying to capture here is something designer friendly, so for example `border` should be a single line such as a `<fieldset>` with the relevant CSS properties it contains as number input (for width), select dropdown (for style) and autocomplete (for color). It's a similar UI for other design tools like figma.